draft/ZEP0004.md

```diff
@@ -0,0 +1,194 @@
+This is possible today with [Pint](https://pint.readthedocs.io/), [Xarray](https://xarray.pydata.org/), and [Pint-Xarray](https://pint-xarray.readthedocs.io/)
+
+```python
+import xarray as xr
+import pint_xarray
+ds = xr.open_dataset('example_with_units.zarr', engine='zarr').pint.quantify()
+```
+
+Now the variables have been decoded to unit-aware Pint arrays, based on the unit information encoded via the conventions.
+Likewise, we can serialize the data using the convention back to Zarr automatically.
+
+```python
+ds.dequantify().to_zarr("array_with_units.zarr")
+```
+
+None of this requires any awareness of units on part of Zarr itself.
+
+## Backward Compatibility
+
+Conventions by definition are fully backwards and forwards compatible with all versions of Zarr, including both V2 and V3.
+They require no changes to the spec or extensions and can be ignored by Zarr implementations.
+
+## Detailed description
+
+This ZEP itself describes the _process_ by which conventions may be proposed, discussed, accepted, and published by the Zarr community.
+This process is intended to be much more lightweight and informal than a spec change, which requires a ZEP.
+
+If this ZEP is accepted, Conventions will be added to the <https://zarr-specs.readthedocs.io/> website as a new top-level heading
+and a corresponding folder will be created in the `zarr-developers/zarr-specs` repo.
+A conventions template, which accompanies this ZEP, will be added to that folder.
+
+### New Convention Process
+
+New conventions are proposed via a pull-request to the `zarr-specs` repo which adds a new conventions document.
+If the convention is already documented elsewhere, the convention document can just contain a reference to the external documentation.
+The author of the PR is expected to convene the relevant domain community to review and discuss the ZEP.
+This includes posting a link to the PR on relevant forums, mailing lists, and social-media platforms.
+
+The goal of the discussion is to reach a _consensus_ among the domain community regarding the convention.
+The Zarr steering council, together with the PR author, will determine if a consensus has been reached, at which point the PR
+can be merged and the convention published on the website.
+If a consensus cannot be reached, the steering council may still decide to publish the convention, accompanied by a
+disclaimer that it is not a consensus, and noting any objections that were raised during the discussion.
+
+It is also possible that multiple, competing conventions exist in the same domain. While not ideal, it's not up to
+the Zarr community to resolve such domain-specific debates.
+These conventions should still be documented in a central location, which hopefully helps move towards alignment.
+
+### Updating a Convention
+
+Conventions should be versioned using incremental integers, starting from 1.
```

Comment:
For me, reading these two sentences leaves a certain tension. Can we clarify if the first is really a SHOULD or is it more a MAY?